bottle_app.py

```diff
@@ -629,14 +629,59 @@ def api_get_movie_trackpoints():
         labels         = sorted( ( tp['label'] for tp in trackpoint_dicts) )
         frame_dicts    = defaultdict(dict)
 
+        ruler0_x, ruler20_x = -1, -1
+        ruler0_y, ruler20_y = -1, -1
+        ruler0_frame_number = -1
+        sensor_width_mm = 36  # Sensor width 36 millimeters (this is a common value,
+        # but should be adjusted according to actual circumstances)
+        actual_distance_mm = 20
+        mm_per_inch = 25.4  # common value
+
         for tp in trackpoint_dicts:
-            frame_dicts[tp['frame_number']][tp['label']+' x'] = tp['x']
-            frame_dicts[tp['frame_number']][tp['label']+' y'] = tp['y']
+            frame_dicts[tp['frame_number']][tp['label'] + ' x'] = tp['x']
+            frame_dicts[tp['frame_number']][tp['label'] + ' y'] = tp['y']
+            if tp['label'] == 'ruller 0 mm':
+                ruler0_x = tp['x']
+                ruler0_y = tp['y']
+                ruler0_frame_number = tp['frame_number']
```

Comment:
Ditto here. You're combining too much in this one function. We need to break it up so that it is easier to understand and easier to test.